### DISCUSSION TITLE
[SOLVED] How to pass api key properly?

### DISCUSSION THREAD

**Discussion by KendrickMichaud** [exadel-inc/CompreFace#972] [Q&A]
I don't know if this is an issue with CompreFace or something with the postman requests.

I have built CompreFace on an ec2 instance and have been trying to get postman to work with the API.

I have the API key and all the headers/body but everytime I send the request I get

{
  "message" : "Service API key should be UUID",
  "code" : 26
}
I am sending the x-api-key as:

"cc555555cc5-c555-5555-555555"

Any thoughts or suggestions?

Edit: I tinkered through Postman and disabled every header except for postmantoken, host, content-type, x-api-key. I then removed the quotes and this time it worked!

**Comment by elektrinis** [comment]
Sorry for bringing this up, but I can't figure it out... Where do you find the API key? I'm running HAOS addon and the key is nowhere to be found.

**Reply by hashimkhanzada** [reply]
You'll have to manually get it from the built-in portal. Eg if it's hosted on localhost:8000 (check the docker compose file to see the port it's on) you'd have to go through the sign up process (creates an admin account), then once you create a new service in the portal it will show an api key.

I haven't used Home Assistant so it could be different there.

**Reply by elektrinis** [reply]
Thank you. I could not find this mentioned in documentation.
